### DISCUSSION TITLE
How to detect stops by time, not location?

### DISCUSSION THREAD

**Discussion by robhawkes** [anitagraser/movingpandas#216] [Q&A] [answered]
Hello,

I've been keeping a keen eye on MovingPandas for some time now so I've been excited to have an excuse to use it recently.

I'm currently analysing some bus trajectory data for London, which comes in the format of a state update per vehicle at ~1-minute intervals per vehicle. However no state updates are sent when the vehicle is switched off at the depot overnight. In this case there would be one update when it is switched off at the depot, and then another update some hours later when it is switched on in the morning. It's also possible that (for whatever reason) a vehicle may send its last state update some distance before it reaches the depot, and send its first update in the morning some distance after leaving the depot.

What I'd like to do is detect and plot these moments where vehicles are likely at the depots due to there being >1 hour between updates. I tried using `TrajectoryStopDetector` but this requires a `max_diameter` parameter so it's not suitable because vehicles may have sent their last / first updates some distance before reaching the depot. If I raise the distance far enough to account for this then the function thinks there are stops all along the trajectory so it's full of false positives.

Is there a way to detect stop points based purely on a temporal difference between state updates? Ideally I'd like a way to have the same output as `TrajectoryStopDetector` but only using `min_duration` and not taking into account any spatial distance between states. Eg. Show me all the points where the vehicle didn't send state updates for more than an hour, regardless of how far apart those states are spatially.

Hopefully that makes sense! 😅

Thanks,
Robin

**Comment by anitagraser** [comment, marked as answer]
Hi Robin,
If I understand correctly, the [ObservationGapSplitter](https://movingpandas.readthedocs.io/en/master/trajectorysplitter.html#movingpandas.ObservationGapSplitter) should do the trick. It takes the original multi-day trajectory and splits it into subtrajectories whenever there is an observation gap, i.e. because the transponder is off while in the depot. Then you could use the start and/or end locations of the resulting subtrajectories.

**Reply by robhawkes** [reply]
Thanks for getting back to me about this.

It looks like `ObservationGapSplitter` will answer the bulk of the question, that is breaking the trajectories up so I can differentiate the points that have a specific time delta between them (the start/end points you mention).

I would still like to retain the time delta as a value on each of those points (eg. to plot a circle with size relative to time delta) so I'll take a look at whether `ObservationGapSplitter` retains this or if I'll need to calculate it myself. I have an SQLite query that logs the time delta now so it's less integral for me that MovingPandas can do every part of this process.
